azure-kusto-ingest/azure/kusto/ingest/ingestion_properties.py

```diff
@@ -69,15 +71,112 @@ class ReportMethod(IntEnum):
     Queue = 0
 
 
+class MappingConsts(Enum):
+    """Mapping properties keys."""
+    # Json Mapping consts
+    PATH = "Path",
+    TRANSFORMATION_METHOD = "Transform"
+    # csv Mapping consts
+    ORDINAL = "Ordinal"
+    CONST_VALUE = "ConstValue"
+    # Avro Mapping consts
+    FIELD_NAME = "Field"
+    COLUMNS = "Columns"
+    # General Mapping consts
+    STORAGE_DATA_TYPE = "StorageDataType"
+
+
+class TransformationMethod(Enum):
+    # Transformations to configure over json column mapping
+    NONE = "None"
+    PROPERTY_BAG_ARRAY_TO_DICTIONARY = "PropertyBagArrayToDictionary",
+    SOURCE_LOCATION = "SourceLocation"
+    SOURCE_LINE_NUMBER = "SourceLineNumber"
+    GET_PATH_ELEMENT = "GetPathElement"
+    UNKNOWN_ERROR = "UnknownMethod"
+    DATE_TIME_FROM_UNIX_SECONDS = "DateTimeFromUnixSeconds"
+    DATE_TIME_FROM_UNIX_MILLISECONDS = "DateTimeFromUnixMilliseconds"
+    DATE_TIME_FROM_UNIX_MICROSECONDS = "DateTimeFromUnixMicroseconds"
+    DATE_TIME_FROM_UNIX_NANOSECONDS = "DateTimeFromUnixNanoseconds"
+
+
 class ColumnMapping:
-    """Abstract class to column mapping."""
+    """ Common class to all the column mapping kinds."""
+
+    def __init__(self, columnName, cslDataType, properties={}):
+        self.columnName = columnName
+        self.columnType = cslDataType
+        self.properties = properties
+
+    def setPath(self, path):
+        self.properties[MappingConsts.PATH.value] = path
+
+    def getPath(self):
+        return self.properties[MappingConsts.PATH.value]
+
+    def setTransform(self, transform=TransformationMethod.NONE):
+        self.properties[MappingConsts.TRANSFORMATION_METHOD.value] = transform.value
+
+    def getTransform(self):
+        return self.properties[MappingConsts.TRANSFORMATION_METHOD.value]
+
+    def setOrdinal(self, ordinal):
+        self.properties[MappingConsts.ORDINAL.value] = str(ordinal)
+
+    def getOrdinal(self):
+        return int(self.properties[MappingConsts.ORDINAL.value])
+
+    def setConstantValue(self, constValue):
+        self.properties[MappingConsts.CONST_VALUE.value] = constValue
+
+    def getConstantValue(self):
+        return self.properties[MappingConsts.CONST_VALUE.value]
+
+    def setField(self, field):
+        self.properties[MappingConsts.FIELD_NAME.value] = field
+
+    def setField(self):
+        return self.properties[MappingConsts.FIELD_NAME.value]
+
+    def setColumns(self, columns):
+        self.properties = [MappingConsts.COLUMNS.value] = columns
+
+    def getColumns(self):
+        return self.properties[MappingConsts.COLUMNS.value]
+
+    def setStorageDataType(self, dataType):
+        self.properties[MappingConsts.STORAGE_DATA_TYPE.value] = dataType
+
+    def getStorageDataType(self):
+        return self.properties[MappingConsts.STORAGE_DATA_TYPE.value]
+
+    def isValid(self, ingestionMappingType):
```

Comment:
Maybe add a unit test

Reply:
i will actually remove this - as it was suggested in the .Net that this only covers a small part of the cases and might not belong here